applications/hotornot/src/hon_util.erl

```diff
@@ -103,53 +107,73 @@ build_keys(<<D:1/binary, Rest/binary>>, Prefix, Acc) ->
     build_keys(Rest, <<Prefix/binary, D/binary>>, [kz_util:to_integer(<<Prefix/binary, D/binary>>) | Acc]);
 build_keys(<<>>, _, Acc) -> Acc.
 
-%% Given a list of rates, return the list of rates whose routes regexes match the given E164
-%% Optionally include direction of the call and options from the client to match against the rate
--spec matching_rates(kz_json:objects(), ne_binary()) ->
+-spec matching_rates(kz_json:objects(), kz_json:object()) ->
                             kz_json:objects().
--spec matching_rates(kz_json:objects(), ne_binary(), api_binary(), trunking_options()) ->
-                            kz_json:objects().
-matching_rates(Rates, DID) ->
-    matching_rates(Rates, DID, 'undefined', []).
-
-matching_rates(Rates, DID, Direction, RouteOptions) ->
-    E164 = knm_converters:normalize(DID),
-    [Rate || Rate <- Rates,
-             matching_rate(Rate, E164, Direction, RouteOptions)
-    ].
+matching_rates(Rates, ReqJObj) ->
+    FilterList = kapps_config:get(?APP_NAME, <<"filter_list">>, ?DEFAULT_FILTER_LIST),
+    lists:foldl(fun(Filter, Acc) ->
+                        lists:filter(fun(R) -> matching_rate(R, Filter, ReqJObj) end, Acc)
```

Comment:
There could be N matching rates by prefix. `FilterList` I believe then allows you to filter those rates that matched by prefix by more criteria. Then, once you have the final list of "matching" rates, you sort them and take the first one in the list.